Modules/Networking/Core/FixedSizeJSONRequest.swift

```diff
@@ -0,0 +1,149 @@
+  private static var paddingParameterName: String { "padding" }
+
+  /// The additional overhead caused by adding a field with the given `paddingParameterName`
+  private static var paddingParameterOverhead: Int { "\"\(Self.paddingParameterName)\":\"\"".utf8.count }
+
+  /// The target size that this `ParameterEncoding` must guarantee
+  var targetSize: Int
+
+  /// Given a dictionary of parameters and a given target size, adds an additional `padding` parameter large enough so it adds
+  /// exactly `size` bytes to the JSON representation of the given `parameters`.
+  private static func addPadding(to parameters: [String: Any], size: Int) throws -> [String: Any] {
+    var paddedParameters = parameters
+    let paddingOverhead = parameters.isEmpty
+      ? Self.paddingParameterOverhead
+      : Self.paddingParameterOverhead + 1 // accounts for the additional "," character
+
+    let paddingSize = size - paddingOverhead
+
+    guard paddingSize >= 0 else {
+      /// There is no way of fitting an additional parameter
+      throw Error.paddingOverflow
+    }
+
+    if paddingSize > 0 {
+      paddedParameters[Self.paddingParameterName] = String.random(size: paddingSize)
+    }
+    return paddedParameters
+  }
+
+  /// Note that compared to the implementation of `JSONEncoding` provider by Alamofire, this encoder purposefully avoids adding
+  /// the `Content-Type` header, which then must be explicitly added to the request on the application layer.
+  func encode(_ urlRequest: URLRequestConvertible, with parameters: Parameters?) throws -> URLRequest {
+    var urlRequest = try urlRequest.asURLRequest()
+
+    let urlSize = urlRequest.url?.byteCount ?? 0
+    let methodSize = urlRequest.method?.byteCount ?? 0
+    let headersSize = urlRequest.headers.map { $0.byteCount }.reduce(0, +)
+
+    let originalParameters = parameters ?? [:]
+    let originalParametersSize = originalParameters.httpRepresentation.byteCount
+
+    let originalRequestSize = urlSize + methodSize + headersSize + originalParametersSize
+
+    let totalPaddingSize = self.targetSize - originalRequestSize
+    let paddedParameters = try Self.addPadding(to: originalParameters, size: totalPaddingSize)
+
+    let data = try JSONSerialization.data(withJSONObject: paddedParameters)
+    urlRequest.httpBody = data
+
+    return urlRequest
+  }
+
+  enum Error: Swift.Error {
+    /// The request is already too big.
+    case paddingOverflow
+  }
+}
+
+// MARK: - HTTPRepresentable
+
+/// A protocol that describes something that can be represented with a given String in a raw HTTP/1.1 request
+protocol HTTPRepresentable {
+  /// The raw representation of the object when encoded in an HTTP/1.1 request
+  var httpRepresentation: String { get }
+}
+
+extension HTTPRepresentable {
+  /// The size in bytes of the `httpRepresentation` for this object
+  var byteCount: Int {
+    return self.httpRepresentation.utf8.count
+  }
+}
+
+extension Parameters {
+  var httpRepresentation: String {
```

Comment:
is this correct? Aren't parameters also serialized in other ways? (e.g., query string)

Reply:
Yes, but `JSONRequest`, which is extended by `FixedSizeJSONRequest`, already assumes that the HTTP method of the request sends its parameters in the body.

Reply:
Right. GTM